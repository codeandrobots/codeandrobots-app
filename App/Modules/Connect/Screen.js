import React, { Component } from 'react'
import { View, ScrollView, Text } from 'react-native'
import PropTypes from 'prop-types'

import Types from 'App/Services/PropTypes'

import {
  Container,
  TextInput,
  Footer,
  Card,
  List,
  ListItem,
  Modal } from 'App/Components'

import { Metrics, Images } from 'App/Themes'

import s from './Styles'

export default class Screen extends Component {
  static propTypes = {
    robot: PropTypes.string,
    error: Types.error,
    connectTo: PropTypes.string,
    enabled: PropTypes.bool.isRequired,
    scanning: PropTypes.bool.isRequired,
    devices: PropTypes.arrayOf(Types.bluetoothDevice).isRequired,
    activeDevice: Types.bluetoothDevice,
    showProblemsConnectingModal: PropTypes.bool.isRequired,
    showIsYourDeviceSupportedModal: PropTypes.bool.isRequired,
    instructions: PropTypes.string.isRequired,
    networkAdded: PropTypes.bool.isRequired,
    ssid: PropTypes.string,
    password: PropTypes.string,
    host: PropTypes.string,
    port: PropTypes.number,
    onEnableBluetooth: PropTypes.func.isRequired,
    onScan: PropTypes.func.isRequired,
    onConnect: PropTypes.func.isRequired,
    onDisconnect: PropTypes.func.isRequired,
    onDone: PropTypes.func.isRequired,
    onEmailInstructions: PropTypes.func.isRequired,
    onIsYourDeviceSupported: PropTypes.func.isRequired,
    onProblemsConnecting: PropTypes.func.isRequired,
    onHideProblemsConnectingModal: PropTypes.func.isRequired,
    onChangeText: PropTypes.func.isRequired
  }

  renderConnectTo = () => {
    const { onConnectTo, onIsYourDeviceSupported } = this.props
    return (
      <Container>
        <Card
          image={Images.bluetooth}
          link={'Is the device you want\nto connect to supported?'}
          onLinkPress={onIsYourDeviceSupported} />
        {this.renderIsYourDeviceSupportedModal()}
        <Footer style={{paddingTop: 0}}>
          <Card
            title='CHOOSE HOW TO CONNECT'
            text='Connect to a bluetooth device OR to the Code & Robots simulator'
            button='Connect Bluetooth Device'
            link={'Connect to the\nCode & Robots Simulator'}
            textAlign='left'
            onPress={() => { onConnectTo('device') }}
            onLinkPress={() => { onConnectTo('simulator') }}
            style={{textView: {marginBottom: Metrics.unit * 2}}} />
        </Footer>
      </Container>
    )
  }

  renderConnectedToSimulator = () => {
    const { instructions, onDone, onEmailInstructions } = this.props
    return (
      <Container>
        <Footer style={{paddingTop: 0}}>
          <Card
            title='YOU`RE CONNECTED TO THE SIMULATOR'
            text={instructions}
            button='Done'
            link='Email simulator link'
            textAlign='left'
            onPress={onDone}
            onLinkPress={onEmailInstructions}
            style={{textView: {marginBottom: Metrics.unit * 2}}} />
        </Footer>
        <Card
          image={Images.bluetooth}
          title='Good job 👍' />
      </Container>
    )
  }

  renderAddNetwork = () => {
    const {
      ssid,
      password,
      onChangeText,
      onAddNetwork } = this.props

    return (
      <Container>
        <Footer style={{paddingTop: 0}}>
          <Card
            title='ADD WIFI NETWORK'
            text='Add wifi network that this device is connected to'
            button='Add'
            textAlign='left'
            onPress={onAddNetwork}
            style={{textView: {marginBottom: Metrics.unit * 2}}} />
        </Footer>
        <View style={s.formView}>
          <Text style={s.formTitle}>Add network</Text>
          <View style={s.form}>
            <TextInput
              style={s.input}
              name='SSID'
              placeholder='SSID'
              value={ssid}
              onChangeText={(value) => { onChangeText('ssid', value) }}
            />
            <TextInput
              style={s.input}
              name='Password'
              placeholder='Password'
              value={password}
              secureTextEntry
              onChangeText={(value) => { onChangeText('password', value) }}
            />
          </View>
        </View>
      </Container>
    )
  }

  renderConnectTheMark = () => {
    const {
      ssid,
      password,
      host,
      port,
      onDone } = this.props

    const qr = {
      ssid,
      password,
      host,
      port
    }
    return (
      <Container>
        <Card
          qr={JSON.stringify(qr)}
          title='Scan this QR using the MARK camera'
          text={`or connect to ${host}:${port}`}
        />
        <Footer style={{paddingTop: 0}}>
          <Card
            title='CONNECT THE MARK'
            text='Tap done once connected'
            button='Done'
            textAlign='left'
            onPress={onDone}
            style={{textView: {marginBottom: Metrics.unit * 2}}} />
        </Footer>
      </Container>
    )
  }

  renderTurnOnBluetooth = () => {
    const { onEnableBluetooth, onProblemsConnecting } = this.props
    return (
      <Container>
        <Footer style={{paddingTop: 0}}>
          <Card
            title='BLUETOOTH IS TURNED OFF'
            text='You need to turn on Bluetooth before connecting'
            button='Turn on Bluetooth'
            link='Problems Connecting?'
            textAlign='left'
            onPress={onEnableBluetooth}
            onLinkPress={onProblemsConnecting}
            style={{textView: {marginBottom: Metrics.unit * 2}}} />
        </Footer>
        <Card image={Images.bluetooth} />
        {this.renderProblemsConnectingModal()}
      </Container>
    )
  }

  renderConnectedDevice = () => {
    const {
      activeDevice,
      onDone,
      onDisconnect } = this.props

    return (
      <Container>
        <List>
          <ListItem
            key={activeDevice.id}
            title={activeDevice.name}
            text='Connected, tap to disconnect'
            buttonIcon='bluetooth-b'
            buttonIconSize={22}
            onPress={() => { onDisconnect() }} />
        </List>
        <Footer style={{paddingTop: 0}}>
          <Card
            title='YOU’RE CONNECTED'
            text='All hooked up & ready to go!'
            button='Done'
            textAlign='left'
            onPress={onDone}
            style={{textView: {marginBottom: Metrics.unit * 2}}} />
        </Footer>
      </Container>
    )
  }

  renderDevices = () => {
    const {
      enabled,
      scanning,
      devices,
      activeDevice,
      onScan,
      onConnect,
      onProblemsConnecting } = this.props

    const footerText = (scanning)
      ? 'Scanning for nearby bluetooth devices...\n'
      : 'Tap on a bluetooth device above to connect or scan to find other devices nearby.'

    return (
      <Container>
        <Footer style={{paddingTop: 0}}>
          <Card
            title='GET CONNECTED'
            text={footerText}
            button='Scan'
            link='Problems Connecting?'
            textAlign='left'
            loading={scanning}
            onPress={onScan}
            onLinkPress={onProblemsConnecting}
            style={{textView: {marginBottom: Metrics.unit * 2}}} />
        </Footer>
        {enabled && (
          <List style={{ marginBottom: 235 }} scrollable>
            {devices.map(device => {
              let text = null
              if (activeDevice && activeDevice.id === device.id) {
                if (activeDevice.isConnecting) {
                  text = 'Connecting...'
                } else if (activeDevice.isConnected) {
                  text = 'Connected, tap to disconnect'
                } else if (activeDevice.error) {
                  text = 'Failed to connect, tap to try again'
                }
              }
              return (
                <ListItem
                  key={device.id}
                  title={device.name}
                  text={text}
                  buttonIcon='bluetooth-b'
                  buttonIconSize={22}
                  onPress={() => { onConnect(device) }} />
              )
            })}
          </List>
        )}
        {this.renderProblemsConnectingModal()}
      </Container>
    )
  }

  renderProblemsConnectingModal = () => {
    const {
      showProblemsConnectingModal,
      onHideProblemsConnectingModal } = this.props
    return <Modal
      navigation={this.props.navigation}
      show={showProblemsConnectingModal}
      onHidePress={onHideProblemsConnectingModal}
      template='ProblemsConnecting' />
  }

  renderIsYourDeviceSupportedModal = () => {
    const {
      showIsYourDeviceSupportedModal,
      onHideIsYourDeviceSupportedModal } = this.props
    return <Modal
      navigation={this.props.navigation}
      show={showIsYourDeviceSupportedModal}
      onHidePress={onHideIsYourDeviceSupportedModal}
      template='IsYourDeviceSupported' />
  }

  render () {
    const {
      connectTo,
      enabled,
      activeDevice,
      networkAdded } = this.props

    if (!connectTo) {
      return this.renderConnectTo()
    }

    if (connectTo === 'device') {
      if (!enabled) {
        return this.renderTurnOnBluetooth()
      }

      if (activeDevice && activeDevice.isConnected) {
        return this.renderConnectedDevice()
      }

      return this.renderDevices()
    }

    if (connectTo === 'simulator') {
      return this.renderConnectedToSimulator()
    }

    if (connectTo === 'mark') {
      if (networkAdded) {
        return this.renderConnectTheMark()
      } else {
        return this.renderAddNetwork()
      }
    }
  }
}
