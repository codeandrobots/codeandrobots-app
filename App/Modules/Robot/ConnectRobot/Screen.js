import React, { Component } from 'react'
import { ScrollView } from 'react-native'
import PropTypes from 'prop-types'

import { Metrics, Fonts } from 'App/Themes'
import {
  Container,
  Separator,
  Card,
  Link,
  Footer,
  Links,
  List,
  CompactListItem } from 'App/Components'

export default class ConnectRobotScreen extends Component {
  static propTypes = {
    type: PropTypes.string,
    image: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
    video: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
    title: PropTypes.string.isRequired,
    text: PropTypes.string,
    links: PropTypes.arrayOf(PropTypes.shape({
      title: PropTypes.string.isRequired,
      url: PropTypes.string.isRequired
    })),
    onLinkPress: PropTypes.func.isRequired,
    onNamePress: PropTypes.func.isRequired,
    onChangePicturePress: PropTypes.func.isRequired,
    onSetupPress: PropTypes.func.isRequired,
    onConnectPress: PropTypes.func.isRequired
  }

  render () {
    const {
      type,
      image,
      video,
      title,
      text,
      links,
      onLinkPress,
      onNamePress,
      onChangePicturePress,
      onSetupPress,
      onConnectPress
    } = this.props

    const isCustomRobot = (type === 'custom')

    const cardTitle = (!isCustomRobot) ? title : null

    return (
      <Container>
        <ScrollView>
          <Card
            style={{
              textView: {
                marginBottom: 0
              }
            }}
            textAlign='left'
            image={image}
            video={video}
            title={cardTitle}
            text={text}
          />
          {!isCustomRobot && <Separator />}
          {links &&
            <Links>
              {links.map((link, i) => {
                return (
                  <Link
                    key={i}
                    style={{
                      view: {
                        marginVertical: Metrics.unit / 2
                      },
                      text: {
                        fontSize: Fonts.size.regular
                      }
                    }}
                    text={link.title}
                    centered
                    uppercase={false}
                    onPress={() => { onLinkPress(link) }}
                  />
                )
              })}
            </Links>
          }
          {isCustomRobot &&
            <List style={{marginHorizontal: Metrics.unit}}>
              <CompactListItem title={title} onPress={onNamePress} />
              <CompactListItem title='Change picture' onPress={onChangePicturePress} />
              <CompactListItem title='Setup' onPress={onSetupPress} />
            </List>
          }
        </ScrollView>
        <Footer style={{ paddingTop: 34 }}>
          <Card
            button='Connect'
            onPress={onConnectPress}
          />
        </Footer>
      </Container>
    )
  }
}
