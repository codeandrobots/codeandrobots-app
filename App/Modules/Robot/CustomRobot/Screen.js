import React, { Component } from 'react'
import { ScrollView } from 'react-native'
import PropTypes from 'prop-types'

import { Metrics } from 'App/Themes'
import {
  Container,
  Separator,
  Card,
  Link,
  Footer,
  Links } from 'App/Components'

export default class CustomRobotScreen extends Component {
  static propTypes = {
    image: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
    video: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
    title: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    links: PropTypes.arrayOf(PropTypes.shape({
      title: PropTypes.string.isRequired,
      config: PropTypes.object.isRequired
    })),
    onLinkPress: PropTypes.func.isRequired
  }

  render () {
    const {
      image,
      video,
      title,
      text,
      links,
      onLinkPress
    } = this.props

    return (
      <Container>
        <ScrollView>
          <Card
            style={{
              textView: {
                marginBottom: 0
              }
            }}
            textAlign='left'
            image={image}
            video={video}
            title={title}
            text={text}
          />
          <Separator />
          {links &&
            <Links>
              {links.map((link, i) => {
                return (
                  <Link
                    key={i}
                    text={link.title}
                    centered
                    uppercase={false}
                    onPress={() => { onLinkPress(link) }}
                  />
                )
              })}
            </Links>
          }
        </ScrollView>
        <Footer style={{ paddingTop: 34 }}>
          <Card
            title='ADD A CUSTOM ROBOT'
            text={'Choose between adding a custom robot from scratch or based on the OttoDIY or Nybble robots.'}
            textAlign='left'
            style={{textView: {marginBottom: Metrics.unit}}} />
        </Footer>
      </Container>
    )
  }
}
