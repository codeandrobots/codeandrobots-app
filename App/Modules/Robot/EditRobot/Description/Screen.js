import React, { Component } from 'react'
import { View } from 'react-native'
import PropTypes from 'prop-types'

import {
  Container,
  TextInput } from 'App/Components'

import s from './Styles'

export default class EditRobotDescriptionScreen extends Component {
  static propTypes = {
    description: PropTypes.string,
    onChangeText: PropTypes.func.isRequired
  }

  render () {
    const {
      description,
      onChangeText } = this.props

    return (
      <Container>
        <View style={s.formView}>
          <View style={s.form}>
            <TextInput
              style={s.input}
              name='Robot Description'
              placeholder='Robot Description'
              value={description}
              multiline
              numberOfLines={3}
              onChangeText={(value) => { onChangeText('description', value) }}
            />
          </View>
        </View>
      </Container>
    )
  }
}
