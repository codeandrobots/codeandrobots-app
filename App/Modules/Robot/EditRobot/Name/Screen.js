import React, { Component } from 'react'
import { View } from 'react-native'
import PropTypes from 'prop-types'

import {
  Container,
  TextInput } from 'App/Components'

import s from './Styles'

export default class EditRobotNameScreen extends Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
    onChangeText: PropTypes.func.isRequired
  }

  render () {
    const {
      name,
      onChangeText } = this.props

    return (
      <Container>
        <View style={s.formView}>
          <View style={s.form}>
            <TextInput
              style={s.input}
              name='Robot Name'
              placeholder='Robot Name'
              value={name}
              onChangeText={(value) => { onChangeText('name', value) }}
            />
          </View>
        </View>
      </Container>
    )
  }
}
