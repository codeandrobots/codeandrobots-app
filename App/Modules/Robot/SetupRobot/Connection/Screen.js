import React, { Component } from 'react'
import { View, Text } from 'react-native'
import PropTypes from 'prop-types'

import { Container, List, ListHeader, OptionListItem } from 'App/Components'

import s from './Styles'

export default class SetupRobotConnectionScreen extends Component {
  static propTypes = {
    connection: PropTypes.object.isRequired,
    onChange: PropTypes.func.isRequired
  }

  onNewLineChange = (value) => {
    const { connection, onChange } = this.props
    onChange({
      ...connection,
      settings: {
        ...connection.settings,
        newLine: value
      }
    })
  }

  render () {
    const { connection } = this.props
    const includeNewLine = !!connection.settings && connection.settings.newLine
    return (
      <Container>
        <List>
          <ListHeader title='Bluetooth' />
          <OptionListItem
            title='Include new line'
            value={!includeNewLine}
            selected={includeNewLine}
            onPress={this.onNewLineChange} />
        </List>
        <View style={s.view}>
          <Text style={s.text}>Wifi support coming soon</Text>
        </View>
      </Container>
    )
  }
}
