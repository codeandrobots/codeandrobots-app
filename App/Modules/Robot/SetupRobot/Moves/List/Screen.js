import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { Container, CompactListItem } from 'App/Components'
import { capitalizeFirstLetter } from 'App/Services/TextUtils'

export default class SetupRobotMovesScreen extends Component {
  static propTypes = {
    commands: PropTypes.object.isRequired,
    onCommandPress: PropTypes.func.isRequired
  }

  render () {
    const { commands, onCommandPress } = this.props
    return (
      <Container>
        {commands && Object.entries(commands).map(([command, value]) =>
          <CompactListItem
            key={`${command}-${value}`}
            title={`${capitalizeFirstLetter(command)}`}
            subtitle={`${value || ' '}`}
            onPress={() => {
              onCommandPress(command, value)
            }} />
        )}
      </Container>
    )
  }
}
