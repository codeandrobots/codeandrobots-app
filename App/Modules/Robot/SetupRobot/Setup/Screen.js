import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { Container, CompactListItem } from 'App/Components'
import { capitalizeFirstLetter } from 'App/Services/TextUtils'

export default class SetupRobotScreen extends Component {
  static propTypes = {
    setup: PropTypes.object.isRequired,
    onConnectionPress: PropTypes.func.isRequired,
    onMovesPress: PropTypes.func.isRequired,
    onSkillsPress: PropTypes.func.isRequired
  }

  render () {
    const {
      setup,
      onConnectionPress,
      onMovesPress,
      onSkillsPress } = this.props
    return (
      <Container>
        <CompactListItem title='Connection' subtitle={`${capitalizeFirstLetter(setup.connection.type)}`} onPress={onConnectionPress} />
        <CompactListItem key={`moves-${Object.entries(setup.commands).length}`} title='Moves' subtitle={`${Object.entries(setup.commands).length}`} onPress={onMovesPress} />
        <CompactListItem key={`commands-${Object.entries(setup.skills).length}`} title='Commands' subtitle={`${setup.skills.length}`} onPress={onSkillsPress} />
      </Container>
    )
  }
}
