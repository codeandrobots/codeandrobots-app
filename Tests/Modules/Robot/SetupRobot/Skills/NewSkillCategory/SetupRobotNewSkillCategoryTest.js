import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Screen from 'App/Modules/Robot/SetupRobot/Skills/NewSkillCategory/Screen'

describe('SetupRobotNewSkillCategory', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <Screen
        category='Sounds'
        onChange={() => {}} />)
  })

  it('Component renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
