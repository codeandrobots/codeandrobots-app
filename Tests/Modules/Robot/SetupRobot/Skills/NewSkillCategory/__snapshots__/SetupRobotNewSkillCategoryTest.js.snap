// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetupRobotNewSkillCategory Component renders correctly 1`] = `
<Container>
  <View
    style={
      Object {
        "margin": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#354052",
          "fontFamily": "Roboto-Medium",
          "fontSize": 18,
        }
      }
    >
      Category
    </Text>
    <View
      style={
        Object {
          "marginVertical": 8,
        }
      }
    >
      <TextInput
        name="Sounds"
        onChangeText={[Function]}
        style={
          Object {
            "marginVertical": 4,
          }
        }
        value="Sounds"
      />
    </View>
  </View>
</Container>
`;
